lib/drivers/i-sens/careSens.js

```diff
@@ -0,0 +1,604 @@
+          annotate.annotateEvent(recordBuilder, annotation);
+        }
+
+        const postRecord = recordBuilder.done();
+        delete postRecord.index;
+        data.post_records.push(postRecord);
+      } else if (hasFlag(FLAGS.CONTROL_SOLUTION, record.flags)) {
+        debug('Skipping BG control solution test');
+      }
+    });
+  };
+
+  const buildKetoneRecords = (data) => {
+    _.forEach(data.records, (record, index) => {
+      if (hasFlag(FLAGS.KETONE, record.flags)) {
+        let { value } = record;
+
+        // According to spec, HI > 8 mmol/L
+        // there is no LO as values are between 0 and 8 mmol/L
+        let annotation = null;
+        if (hasFlag(FLAGS.HI, record.flags)) {
+          value = KETONE_HI + (1 / KETONE_VALUE_FACTOR);
+          annotation = {
+            code: 'ketone/out-of-range',
+            value: 'high',
+            threshold: KETONE_HI,
+          };
+        } else {
+          value = _.toInteger(value) / KETONE_VALUE_FACTOR;
+        }
+
+        if (!hasFlag(FLAGS.CONTROL_SOLUTION, record.flags)) {
+          const recordBuilder = cfg.builder.makeBloodKetone()
+            .with_value(value)
+            .with_units('mmol/L') // values are always in 'mmol/L'
+            .with_deviceTime(sundial.formatDeviceTime(record.jsDate))
+            .set('index', index);
+
+          cfg.tzoUtil.fillInUTCInfo(recordBuilder, record.jsDate);
+
+          if (annotation) {
+            annotate.annotateEvent(recordBuilder, annotation);
+          }
+
+          const postRecord = recordBuilder.done();
+          delete postRecord.index;
+          data.post_records.push(postRecord);
+        } else {
+          debug('Skipping ketone control solution test');
+        }
+      }
+    });
+  };
+
+  return {
+    /* eslint no-param-reassign:
+       [ "error", { "props": true, "ignorePropertyModificationsFor": ["data"] } ] */
+
+    detect(deviceInfo, cb) {
+      debug('no detect function needed', deviceInfo);
+      cb(null, deviceInfo);
+    },
+
+    setup(deviceInfo, progress, cb) {
+      debug('in setup!');
+      progress(100);
+      cb(null, { deviceInfo });
+    },
+
+    connect(progress, data, cb) {
+      hidDevice.connect(cfg.deviceInfo, CareSens.probe, (err) => {
+        if (err) {
+          cb(err);
+        } else {
+          (async () => {
+            CareSens.enableUART(hidDevice);
+            await driver.ping();
+
+            data.disconnect = false;
+            progress(100);
+            cb(null, data);
+          })().catch((error) => cb(error));
+        }
+      });
+    },
+
+    getConfigInfo(progress, data, cb) {
+      progress(0);
+
+      (async () => {
+        cfg.deviceInfo.serialNumber = await driver.getSerialNumber();
+        cfg.deviceInfo.deviceId = `${cfg.deviceInfo.driverId}-${cfg.deviceInfo.serialNumber}`;
+        cfg.deviceInfo.deviceTime = sundial.formatDeviceTime(await driver.getDateTime());
+        debug('Config:', cfg);
+      })().then(() => {
```

Comment:
Do you need a `.then` here? You're already in an `async` block, so the code in this block could just be in the `async` block, no?

Reply:
Good point! I'll remove it.